sale_commission/models/settlement.py

```diff
@@ -121,15 +122,26 @@ def _add_extra_invoice_lines(self, settlement):
         return []
 
     @api.multi
-    def make_invoices(self, journal, product, date=False):
+    def make_invoices(self, journal, refund_journal, product, date=False):
         invoice_obj = self.env['account.invoice']
         for settlement in self:
+            # select the proper journal according to settlement's amount
+            # considering _add_extra_invoice_lines sum of values
+            extra_invoice_lines = self._add_extra_invoice_lines(settlement)
+            extra_total = sum(x['price_unit'] for x in extra_invoice_lines)
+            invoice_journal = (journal if
+                                (settlement.total + extra_total) >= 0 else
+                                refund_journal)
```

Comment:
And please squash commits in one, although your commits messages are very funny :smiley: